generic-lens/src/Data/Generics/Labels.hs

```diff
@@ -107,15 +120,18 @@ instance {-# INCOHERENT #-} AsConstructor name s t a b => Constructor name s t a
 instance {-# INCOHERENT #-} AsConstructor' name s a => Constructor name s s a a where
   constructorPrism = _Ctor' @name
 
-data LabelType = FieldType | LegacyConstrType | ConstrType
+data LabelType = FieldType | LegacyConstrType | ConstrType | PositionType
 
 type family ClassifyLabel (name :: Symbol) :: LabelType where
   ClassifyLabel name =
-    If (CmpSymbol "_@" name == 'LT && CmpSymbol "_[" name == 'GT)
-      'LegacyConstrType
-      ( If (CmpSymbol "@" name == 'LT && CmpSymbol "[" name == 'GT)
-          'ConstrType
-          'FieldType
+    If (StartsWithDigit name)
+      'PositionType
+      ( If (CmpSymbol "_@" name == 'LT && CmpSymbol "_[" name == 'GT)
```

Comment:
It would be nice if these predicates also had names? I guess the reason you pulled out `StartsWithDigit` is so you can define it conditionally, but it would still make things more consistent to pull out the others too.

Reply:
Sounds good, added a commit doing that :+1: